lib/rubygems/printable_uri.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+require 'uri'
+require_relative 'uri_parser'
+
+class Gem::PrintableUri
+  def self.parse_uri(uri)
+    new(uri).parse_uri
+  end
+
+  def initialize(original_uri)
+    @credential_redacted = false
```

Comment:
What's the purpose of `credential_redacted?`? I don't think it's used anywhere?

Reply:
Done [`9bee1e6` (#4868)](https://github.com/rubygems/rubygems/pull/4868/commits/9bee1e6b0b3531e6412a07337be3611a9250ed54)

Reply:
`redactable_credential?` was unused too, but I meant the whole tracking of whether credentials have been redacted or not. I think `@credential_redacted` seems unnecessary at this point since the whole purpose of this class is to redact credentials? Maybe we can redact credentials at instantiation time and we don't need to track whether we have done that or not at all?

Reply:
Yeah, make sense. I've removed it [`e085076` (#4868)](https://github.com/rubygems/rubygems/pull/4868/commits/e085076be272436af85df1e1a11991758c6ccb4f)